controllers/gatekeeper_controller.go

```diff
@@ -120,10 +154,22 @@ func (r *GatekeeperReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&operatorv1alpha1.Gatekeeper{}).
+		WithEventFilter(predicate.Funcs{
```

Comment:
Is this needed?

Reply:
Yes. This prevents reconciling for certain additional unnecessary reconcile loops for certain types of updates

Reply:
After looking more into this, it's not required, but helps reduce unnecessary noise.